sdk/core/Azure.Core/src/Internal/TypeReferenceTypeAttribute.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Azure.Core
+{
+    /// <summary>
+    /// An attribute class indicating a reference type which can replace a type in target RPs.
```

Comment:
Curious: RPs is a management plane-specific term -- is there a more general use case than this, or would it only ever be used for management plane?

Reply:
👍 We should generalize this desscription.